projects/admin/src/app/record/custom-editor/libraries/library-form.service.ts

```diff
@@ -16,125 +16,166 @@
  */
 
 import { Injectable } from '@angular/core';
-import { AbstractControl, FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { AbstractControl, AbstractControlOptions, FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
 import { RecordService, TimeValidator } from '@rero/ng-core';
-import { Subject } from 'rxjs';
-import { Library, NotificationSettings, NotificationType } from '../../../classes/library';
+import { forkJoin, Subject } from 'rxjs';
+import { AcquisitionInformations, Library, NotificationSettings, NotificationType } from '../../../classes/library';
 import { WeekDays } from '../../../classes/week-days';
 
-
-
 @Injectable({
    providedIn: 'root'
 })
 export class LibraryFormService {
 
+  // SERVICE ATTRIBUTES =======================================================
   /** Angular form group */
   public form;
 
-
-  /** RERO ILS notification types */
+  /** RERO-ILS notification types */
   private notificationTypes = [];
-
+  /** RERO-ILS communication languages */
+  private availableCommunicationLanguages = [];
+  /** RERO-ILS countries */
+  private countryList = [];
   /** Observable for build event */
   private buildEvent = new Subject();
 
-  /** RERO ILS communication languages */
-  private availableCommunicationLanguages = [];
+  // GETTER & SETTER ==========================================================
+  setId(id) { this.form.value.id = id; }
```

Comment:
Why not using the typescript setter for example `set libraryPid()`?

Reply:
This is legacy code (initially at the bottom of the file). Just moving up.
But it seems not used yet... so I will remove it